### DISCUSSION TITLE
Odd Disk Usage Amount

### DISCUSSION THREAD

**Discussion by reefland** [longhorn/longhorn#4422] [Q&A] [answered]
Clearly something I don't understand here.  I defined a small `10Mi` PVC to store a conf file and a small dataset:

```
$ k get pvc -n mosquitto  mosquitto-mqtt-data                                                           
NAME                  STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
mosquitto-mqtt-data   Bound    pvc-a343686e-a0c9-4d0d-9766-554ec9b00828   10Mi       RWO            longhorn       4d17h
```

Inside the container, I'm using 392K of space:
```
# mount | grep data
/dev/longhorn/pvc-a343686e-a0c9-4d0d-9766-554ec9b00828 on /mosquitto/data type ext4 (rw,relatime)

# df -h /mosquitto/data
Filesystem                Size      Used Available Use% Mounted on
/dev/longhorn/pvc-a343686e-a0c9-4d0d-9766-554ec9b00828
                          5.7M    392.0K      5.1M   7% /mosquitto/data
```

It hasn't even provisioned the full 10Mi of space yet.

However, Longhorn dashboard shows it's using 32Mi ?!?
![image](https://user-images.githubusercontent.com/16230914/184696534-bac9144a-840e-4957-bbd2-f8c7d937a6ba.png)

And prometheus is sending out alerts that my volume is 320% of capacity:
```
The actual used space of Longhorn volume is over 90% of the capacity.
The actual space used by Longhorn volume pvc-a343686e-a0c9-4d0d-9766-554ec9b00828 on k3s02 is at 320.3125% capacity for more than 5 minutes.
```

```
longhorn_volume_actual_size_bytes{container="longhorn-manager", endpoint="manager", instance="10.42.1.65:9500", job="longhorn-backend", namespace="longhorn-system", node="k3s02", pod="longhorn-manager-pjjtg", service="longhorn-backend", volume="pvc-a343686e-a0c9-4d0d-9766-554ec9b00828"}

33722368

longhorn_volume_capacity_bytes{container="longhorn-manager", endpoint="manager", instance="10.42.1.65:9500", job="longhorn-backend", namespace="longhorn-system", node="k3s02", pod="longhorn-manager-pjjtg", service="longhorn-backend", volume="pvc-a343686e-a0c9-4d0d-9766-554ec9b00828"}

10485760
```

Why is just 400KB of used disk space generated all this?

**Comment by FaraSys** [comment, marked as answer]
Hi. It's normal behavior.
You need read doc about it. https://longhorn.io/docs/1.3.1/volumes-and-nodes/volume-size/
**Longhorn does not support TRIM/UNMAP operations, so the discard mount option or fstrim in the filesystem layer cannot reclaim the unused blocks.**

**Reply by reefland** [reply]
Thank you for the link.  I did not consider the snapshot history as part of the total usage, so that makes sense for being over capacity. 

Can you clarify that statement about Longhorn will not reuse a block? at all?  a block is one time use only?  I'm trying to understand how Longhorn can be used by anyone long-term if this is true.  The documentation referenced this site (https://www.ogris.de/blkalloc/) which implies this is an EXT4 issue.  I usually use XFS on iSCSI.  Should I consider moving to XFS for Longhorn as well?

**Comment by FaraSys** [comment]
I am a user of this product just like you. I just read that the block is not reused. Perhaps in the future the developers will change the behavior.
As I understand it, it's not about ext4. On any FS will be the same behavior. But I didn't check other FS.

**Comment by derekbit** [comment]
We have a ticket for the space reclaim and trim issue.
You can follow up it.
https://github.com/longhorn/longhorn/issues/836
